TPM-Commands/TPM2_StartAuthSession.md

```diff
@@ -145,6 +165,34 @@ that itself used the EK as its `tpmKey` input.
 > that can be used to satisfy HMAC-based authorization for specific
 > objects.  We will not cover this in detail here.
 
+## Establishing Trust in a TPM
+
+Given a computer that has a discrete TPM, how does software running on
+that computer establish trust in the dTPM?
+
+This is an important question since failure to do this will render the
+computer vulnerable to [certain attacks](https://dolosgroup.io/blog/2021/7/9/from-stolen-laptop-to-inside-the-company-network)
+on it.
+
+Use of encryption sessions is a must.  These must be keyed by using a
+key exchange with a public key of the dTPM's that is accessible to the
+caller.  For example, the dTPM's `EKpub`, or any key object with the
+`decrypt`, `fixedTPM`, and `fixedParent` attributes, but not the
+`stClear` attribute, and preferably a primary.  The caller must reliably
+remember this public key as early as possible.  The caller must also
+validate the dTPM's `EKcert` as early as possible (especially before the
+endorsement hierarchy is made unavailable).
+
+Unless the caller has a priori knowledge of that public key for the dTPM
+prior to the first time the caller speaks to the dTPM, then the caller
+will be vulnerable to the dTPM being replaced.
+
+In an ideal world the BIOS would store this public key in protected
```

Comment:
It wouldn't be a huge step to store the TPM's EKpub in a UEFI nvram variable and to have it measured into the TPM as part of the sealing state, sort of like how the PK and db variables are measured and can be used to ensure that the secure boot mode is unchanged. This would be TOFU, unless the OEM does it in some sort of out-of-band way.

Reply:
Can we just make use of UEFI nvram variables?

Reply:
> This would be TOFU, unless the OEM does it in some sort of out-of-band way.

Yes, ideally there would be a standard UEFI NVRAM variable for this and OEMs would populate it for every board they ship.

In the absence of such a standard the best we can do is TOFU -- learn the `EKpub` the first time and go from there.

Also, for a standard we might want to create a primary -with a set template- in the owner hierarchy and use that public key instead just to avoid issues with Windows or other OSes wanting to lock out the endorsement hierarchy for -you know- privacy.